packages/template-retail-react-app/app/pages/checkout/index.test.js

```diff
@@ -142,6 +144,8 @@ test('Can proceed through checkout steps as guest', async () => {
     // update this object, which essentially mimics a saved basket on the backend.
     let currentBasket = JSON.parse(JSON.stringify(ocapiBasketWithItem))
 
+    jest.spyOn(Auth.prototype, 'login').mockReturnValue(mockedGuestCustomer)
```

Comment:
why is this required?

Reply:
The spyOns are required since there are actually 2 different places where login is called during checkout.
1. On app initialization. This is skipped by the mock isTokenValid always returning true
2. When the useCustomer hook is initialized. Because of the mock isTokenValid in 1, all of these tests began to take the new code path and not actually authenticate. The spyOn forces these attempts to actually log in a mock customer so the rest of the test can proceed normally.